test/e2e/tests/http3.go

```diff
@@ -0,0 +1,200 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	httpv1 "net/http"
+	"testing"
+	"time"
+
+	"github.com/quic-go/quic-go"
+	"github.com/quic-go/quic-go/http3"
+	"github.com/quic-go/quic-go/qlog"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTP3Test)
+}
+
+var HTTP3Test = suite.ConformanceTest{
+	ShortName:   "Http3",
+	Description: "Testing http3 request",
+	Manifests:   []string{"testdata/http3.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("Send http3 request", func(t *testing.T) {
+			namespace := "gateway-conformance-http3"
+			routeNN := types.NamespacedName{Name: "http3-route", Namespace: namespace}
+			gwNN := types.NamespacedName{Name: "http3-gateway", Namespace: namespace}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/get",
+					Headers: map[string]string{
+						"Host": "www.example.com",
+					},
+				},
+				Response: http.Response{
+					StatusCode: 200,
```

Comment:
any other h3 response headers that can be used in validation ?

Reply:
ok, I will add more http3 response headers later